<launch>

	<!-- Agregar camera_info -->
	<node pkg="opt_utils" type="add_camera_info" name="add_camera_info">
	</node>

	
	<!-- Convertir imagen depth a point cloud -->
	<node pkg="nodelet" type="nodelet" name="depth_to_pc" args="standalone depth_image_proc/point_cloud_xyz">
	<remap from="camera_info" to="camera/depth/camera_info" />
	<remap from="image_rect" to="camera/depth/image" />
	<remap from="points" to="camera/depth/points" />
	</node>

	
	<!-- Lanzar rosbag -->
	<param name="use_sim_time" value="true" />
	<arg name="args2" default="--clock -l" />
	<arg name="bagfile" default="/home/enmanuel/bags_VD/VD1.bag" />
	<node pkg="rosbag" type="play" name="rosbag" args="$(arg args2) $(arg bagfile)" output="screen">
	<remap from="camera/rgb/color" to="camera/rgb/image_color" />
	</node>

	<!-- Publicar imagen desde compressed -->
	<arg name="args3" default="compressed in:=/camera/rgb/image_color raw out:=/camera/rgb/image_color" />
	<node pkg="image_transport" type="republish" name="compressedToImage" args="$(arg args3)">
	</node>

	<!-- Imagen de disparidad -->
	<node pkg="nodelet" type="nodelet" name="depth_to_disparity" args="standalone depth_image_proc/disparity">
	<remap from="left/image_rect" to="camera/depth/image" />
	<remap from="right/camera_info" to="camera/depth/camera_info" />
	<remap from="left/disparity" to="camera/depth/disparity" />
	</node> 




</launch>
